src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -1576,4 +1713,184 @@ RppStatus rppt_vignette_gpu(RppPtr_t srcPtr,
+                            RpptDescPtr dstDescPtr,
+                            RppPtr_t interDstPtr,
+                            Rpp32f pixelationPercentage,
+                            RpptROIPtr roiTensorPtrSrc,
+                            RpptRoiType roiType,
+                            rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+
+    if ((srcDescPtr->layout != RpptLayout::NCHW) && (srcDescPtr->layout != RpptLayout::NHWC)) return RPP_ERROR_INVALID_SRC_LAYOUT;
+    if ((dstDescPtr->layout != RpptLayout::NCHW) && (dstDescPtr->layout != RpptLayout::NHWC)) return RPP_ERROR_INVALID_DST_LAYOUT;
+    if (pixelationPercentage < 0 || pixelationPercentage > 100)
+        return RPP_ERROR_INVALID_ARGUMENTS;
+
+    RpptInterpolationType interpolationType = RpptInterpolationType::BILINEAR;
+    RpptDesc interDesc;
+    interDesc = *srcDescPtr;
+    RpptDescPtr interDescPtr = &interDesc;
+
+    RpptImagePatchPtr internalDstImgSizes = reinterpret_cast<RpptImagePatch *>(rpp::deref(rppHandle).GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+    RpptROI *internalRoiTensorPtrSrc = reinterpret_cast<RpptROI *>(internalDstImgSizes + dstDescPtr->n);
+
+    for (int i = 0; i < srcDescPtr->n; i++)
+    {
+        internalRoiTensorPtrSrc[i].xywhROI.roiWidth = roiTensorPtrSrc[i].xywhROI.roiWidth;
+        internalRoiTensorPtrSrc[i].xywhROI.roiHeight = roiTensorPtrSrc[i].xywhROI.roiHeight;
+        internalDstImgSizes[i].width = (roiTensorPtrSrc[i].xywhROI.roiWidth * (100 - pixelationPercentage)) / 100;
+        internalDstImgSizes[i].height = (roiTensorPtrSrc[i].xywhROI.roiHeight * (100 - pixelationPercentage)) / 100;
+    }
+
+    interDescPtr->h = (interDescPtr->h * (100 - pixelationPercentage)) / 100;
+    interDescPtr->w = (((interDescPtr->w * ((100 - pixelationPercentage) / 100) ) / 8 ) * 8) + 8;
+    interDescPtr->strides.nStride = interDescPtr->w * interDescPtr->h * interDescPtr->c;
+    interDescPtr->strides.cStride = (srcDescPtr->layout == RpptLayout::NCHW) ? (interDescPtr->w * interDescPtr->h) : 1;
+    interDescPtr->strides.hStride = (srcDescPtr->layout == RpptLayout::NCHW) ? interDescPtr->w : (interDescPtr->c * interDescPtr->w);
+
+    Rpp64u interBufferSize = static_cast<Rpp64u>(interDescPtr->h) * static_cast<Rpp64u>(interDescPtr->w) * static_cast<Rpp64u>(srcDescPtr->c) * static_cast<Rpp64u>(srcDescPtr->n);
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::U8))
+    {
+        hip_exec_resize_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offsetInBytes,
+                               srcDescPtr,
+                               static_cast<Rpp8u*>(interDstPtr),
+                               interDescPtr,
+                               internalDstImgSizes,
+                               interpolationType,
+                               roiTensorPtrSrc,
+                               roiType,
+                               rpp::deref(rppHandle));
+        hipDeviceSynchronize();
```

Comment:
Please change all these to stream synchronize. Like in some other API

Reply:
Done